packages/strapi-plugin-graphql/services/GraphQL.js

```diff
@@ -17,6 +17,424 @@ const GraphQLJSON = require('graphql-type-json');
 const policyUtils = require('strapi-utils').policy;
 
 module.exports = {
+  /**
+   * Returns all fields of type primitive
+   * 
+   * @returns {Boolean}
+   */
+  isPrimitiveType: (_type) => {
+    const type = _type.replace('!', '');
+    return (
+      type === 'Int' ||
+      type === 'Float' ||
+      type === 'String' ||
+      type === 'Boolean'
+    );
+  },
+
+  /**
+   * Returns all fields that are not of type array
+   * 
+   * @returns {Boolean}
+   * 
+   * @example
+   * 
+   * isNotOfTypeArray([String])
+   * // => false
+   * isNotOfTypeArray(String!)
+   * // => true
+   */
+  isNotOfTypeArray: (_type) => {
+    const arrayRegexp = /(\[\w+\])/;
+    const type = _type.replace('!', '');
+    return !arrayRegexp.test(type);
+  },
+
+  /**
+   * Convert non-primitive type to string (non-primitive types corresponds to a reference to an other model)
+   * 
+   * @returns {String}
+   */
+  extractType: function (_type) {
+    return this.isPrimitiveType(_type)
+      ? _type.replace('!', '')
+      : 'ID';
+  },
+
+  /**
+   * Returns a list of fields that have type included in fieldTypes.
+   */
+  getFieldsByTypes: (fields, typeCheck, returnType) => {
+    return _.reduce(fields, (acc, fieldType, fieldName) => {
```

Comment:
I would like to avoid to use Lodash methods, you can use `Object.keys(fields).reduce()` instead